src/main/java/uk/ac/ebi/ampt2d/commons/accession/rest/BasicRestControllerAdvice.java

```diff
@@ -19,29 +19,62 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.bind.WebDataBinder;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.InitBinder;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionCouldNotBeGeneratedException;
 import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.AccessionIsNotPendingException;
 import uk.ac.ebi.ampt2d.commons.accession.core.exceptions.MissingUnsavedAccessionsException;
 
+import javax.validation.ValidationException;
+
 /**
  * Spring {@link RestControllerAdvice} bean to handle exception from the application at rest level and return
  * a specific response with an error message.
  */
 @RestControllerAdvice(assignableTypes = BasicRestController.class)
 public class BasicRestControllerAdvice {
 
+    @Autowired
+    private CollectionValidator collectionValidator;
+
     private static final Logger logger = LoggerFactory.getLogger(BasicRestControllerAdvice.class);
 
     @ExceptionHandler(value = {AccessionIsNotPendingException.class, AccessionCouldNotBeGeneratedException.class,
             MissingUnsavedAccessionsException.class})
     public ResponseEntity<ErrorMessage> handleInternalServerErrors(Exception ex) {
         logger.error(ex.getMessage(), ex);
-        return new ResponseEntity<>(new ErrorMessage(HttpStatus.INTERNAL_SERVER_ERROR, ex),
-                HttpStatus.INTERNAL_SERVER_ERROR);
+        return buildResponseEntity(new ErrorMessage(HttpStatus.INTERNAL_SERVER_ERROR, ex, ex.getMessage()));
+    }
+
+    @ExceptionHandler(value = HttpMediaTypeNotSupportedException.class)
+    public ResponseEntity<ErrorMessage> handleHttpMediaNotSupported(HttpMediaTypeNotSupportedException ex) {
+        return buildResponseEntity(new ErrorMessage(HttpStatus.UNSUPPORTED_MEDIA_TYPE, ex, ex.getMessage()));
+    }
+
+    @ExceptionHandler(value = ValidationException.class)
+    public ResponseEntity<ErrorMessage> handleValidationException(ValidationException ex) {
+        return buildResponseEntity(new ErrorMessage(HttpStatus.BAD_REQUEST, ex, ex.getMessage()));
+    }
+
+    @ExceptionHandler(value = HttpMessageNotReadableException.class)
+    public ResponseEntity<ErrorMessage> handleHttpMessageNotReadable(HttpMessageNotReadableException ex) {
+        return buildResponseEntity(new ErrorMessage(HttpStatus.BAD_REQUEST, ex, "Please provide accepted values"));
+    }
+
+    private ResponseEntity<ErrorMessage> buildResponseEntity(ErrorMessage errorMessage) {
```

Comment:
Change the method profile to the same of Error message, move the creation of error message to the method to build the response.